hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -92,14 +98,19 @@ public SnapshotDeletingService(long interval, long serviceTimeout,
         serviceTimeout);
     this.ozoneManager = ozoneManager;
     this.omSnapshotManager = ozoneManager.getOmSnapshotManager();
-    this.chainManager = ozoneManager.getSnapshotChainManager();
+    OmMetadataManagerImpl omMetadataManager = (OmMetadataManagerImpl)
+        ozoneManager.getMetadataManager();
+    this.chainManager = omMetadataManager.getSnapshotChainManager();
     this.runCount = new AtomicLong(0);
     this.successRunCount = new AtomicLong(0);
     this.suspended = new AtomicBoolean(false);
     this.conf = ozoneManager.getConfiguration();
     this.snapshotDeletionPerTask = conf
         .getLong(SNAPSHOT_DELETING_LIMIT_PER_TASK,
         SNAPSHOT_DELETING_LIMIT_PER_TASK_DEFAULT);
+    this.keyLimitPerSnapshot = conf.getInt(
+        OZONE_SNAPSHOT_KEY_DELETING_LIMIT_PER_TASK,
+        OZONE_SNAPSHOT_KEY_DELETING_LIMIT_PER_TASK_DEFAULT);
   }
 
   private class SnapshotDeletingTask implements BackgroundTask {
```

Comment:
Ideally we need new test cases in `TestSnapshotDeletingService` to cover the rename test cases here.

Or you can file a separate jira for the test case additions.

Reply:
File a separate JIRA : [HDDS-8207](https://issues.apache.org/jira/browse/HDDS-8207)